src/playwright-api/takeArchive.ts

```diff
@@ -36,12 +42,47 @@ async function takeArchive(
   });
 
   // Serialize and capture the DOM
-  const domSnapshot: serializedNode = await page.evaluate(dedent`
+  const domSnapshot: elementNode = await page.evaluate(dedent`
     ${rrweb};
     rrwebSnapshot.snapshot(document, { noAbsolute: true });
   `);
 
+  // XXX_jwir3: We go through and filter any of these that would have file names that would be too large.
+  const sourceMap: Map<string, string> = new Map<string, string>();
+  if (domSnapshot.childNodes.length !== 0) {
+    shortenFileNameSrc(domSnapshot.childNodes, sourceMap);
+  }
+
   testInfo.attach(name, { contentType, body: JSON.stringify(domSnapshot) });
+
+  return Promise.resolve(sourceMap);
+}
+
+function shortenFileNameSrc(
+  input: Array<serializedNodeWithId>,
+  existingSourceMap: Map<string, string>
+): Map<string, string> {
+  // eslint-disable-next-line no-restricted-syntax
+  for (const nextChildNode of input) {
```

Comment:
It looks like we're walking the entire dom tree twice -- once here and once in `write-archive`. Could we possible get that down to once? Maybe by catching the file name errors when trying to write them to disk rather than looking at the length in this pre-processing step?

Reply:
Hm, I'm not sure we're going to be able to condense it into a single walk. We need the source mappings _before_ we start writing to files, otherwise we won't have the data we need when we go to perform the rewrite as we output the files.